### PR TITLE
Add self-dox warning to social media buttons

### LINKED ISSUE #1450: Information disclosure concern - "Tell them to support 2FA" option
While it's a great idea in general, if a user were to only use this feature to notify their own sites it could easily translate to an inadvertent disclosure of a twitter user's affiliations.

Generally-speaking it would take some work to find out what banks a user might be registered with, whether they have cryptocurrencies, where their data is stored. Without at least a note about information disclosure here, this opens the door for a user to disclose that information to a potential attacker on their own. Knowing those sites don't support 2FA additionally reduces the difficulty threshold for a potential attacker.

### PR DESCRIPTION
**Pull request by Carlgo11**
This PR adds a warning message when hovering over social media contact buttons in categories with the tag `sensitive`.

My IDE also cleaned up a bunch of white spaces in desktop-table.html so I apologize if the diff is hard to read :sweat_smile: 

* Resolves #1450
* Resolves #3812

### CODE REVIEW

**Comment by phallobst** [comment]
Reading this diff on GitHub is easier if you hide whitespace-only changes:
image

**Comment by phallobst** [comment]
I'm not quite sure if only certain categories should be considered "sensitive". Depending on your local laws and regulations, almost any category might contain a website that would qualify.

Similar to the region notice in #6484, a flag in the `localStorage` could indicate whether the user wants to keep this warning in. We wouldn't need cookies to do so.

**Comment by ApCoder123** [comment]
Are we planning to incorporate this into the mobile layout as well? Also, agree that it would be a good idea to allow users to disable the warning.

**Comment by Carlgo11** [comment]
I've addressed the requested changes the following ways:
* Made the tooltip a modal that's displayed on all device types
* Added a local storage variable to only display the warning once
* Switched from a custom Bootstrap v4 to vanilla Bootstrap v5 as the prev version didn't contain modals

File: js/social-media.js [hunk 1]
```diff
@@ -17,27 +14,43 @@ $('.email-button').click(function () {
   window.open('mailto:' + $(this).data('email') + '?subject=' + langs.get(lang));
 })
 
+$('.facebook-button').click(function () {
+  const uri = `https://facebook.com/${(this).data('facebook')}`;
```
Comment (phallobst) [inline js/social-media.js]:
```suggestion
  const uri = `https://facebook.com/${$(this).data('facebook')}`;
```

**Review by phallobst — CHANGES_REQUESTED** [review]
I think that after clicking on "I accept the risk", the modal dialog should also be closed.

**Review by phallobst — APPROVED** [review]

**Review by dsoegijono — APPROVED** [review]

**Pull request merged**